components/eam/src/physics/crm/cloud_diagnostics.F90

```diff
@@ -1,525 +1,374 @@
+  ! real(r8) :: lwc(pcols,pver)         ! Grid box average liquid water content
+
+! old data
+  real(r8) :: tpw    (pcols)          ! total precipitable water
+  real(r8) :: clwpold(pcols,pver)     ! Presribed cloud liq. h2o path
+  real(r8) :: hl     (pcols)          ! Liquid water scale height
+
+  integer :: i,k                      ! loop indexes
+  integer :: ncol, lchnk
+  real(r8) :: rgrav
+
+  real(r8) :: allcld_ice (pcols,pver) ! Convective cloud ice
+  real(r8) :: allcld_liq (pcols,pver) ! Convective cloud liquid
+
+  real(r8) :: effcld(pcols,pver)      ! effective cloud=cld*emis
+
+  logical :: dosw,dolw
+  logical :: use_MMF
+
+  !-----------------------------------------------------------------------------
+  if (.not.do_cld_diag) return
+
+  call phys_getopts(use_MMF_out=use_MMF)
+
+  dosw     = .true.
+  dolw     = .true.
+
+  if (.not.(dosw .or. dolw)) return
+
+  ncol  = state%ncol
+  lchnk = state%lchnk
+
+  itim_old = pbuf_old_tim_idx()
+  call pbuf_get_field(pbuf, cld_idx, cld, start=(/1,1,itim_old/), kount=(/pcols,pver,1/) )
+
+  call pbuf_get_field(pbuf, iclwp_idx, iclwp )
+  call pbuf_get_field(pbuf, iciwp_idx, iciwp )
+  call pbuf_get_field(pbuf, dei_idx, dei )
+  call pbuf_get_field(pbuf, mu_idx, mu )
+  call pbuf_get_field(pbuf, lambda_idx, lambda )
+
+  call outfld('dei_cloud',dei(:,:),pcols,lchnk)
+  call outfld('mu_cloud',mu(:,:),pcols,lchnk)
+  call outfld('lambda_cloud',lambda(:,:),pcols,lchnk)
+
+  ! call pbuf_get_field(pbuf, rei_idx, rei )
+  ! call pbuf_get_field(pbuf, rel_idx, rel )
+  ! call outfld('rel_cloud', rel, pcols, lchnk)
+  ! call outfld('rei_cloud', rei, pcols, lchnk)
+
+  call pbuf_get_field(pbuf, cicewp_idx, cicewp )
+  call pbuf_get_field(pbuf, cliqwp_idx, cliqwp )
+
+  ! Compute liquid and ice water paths
+
+  ! ----------------------------------------------------------- !
+  ! Adjust in-cloud water values to take account of convective  !
+  ! in-cloud water. It is used to calculate the values of       !
+  ! iclwp and iciwp to pass to the radiation.                   !
+  ! ----------------------------------------------------------- !
+  if( conv_water_in_rad /= 0 ) then
+    allcld_ice(:ncol,:) = 0._r8 ! Grid-avg all cloud liquid
+    allcld_liq(:ncol,:) = 0._r8 ! Grid-avg all cloud ice
+
+    call conv_water_4rad( state, pbuf, conv_water_in_rad, allcld_liq, allcld_ice )
+  else
+    allcld_liq(:ncol,top_lev:pver) = state%q(:ncol,top_lev:pver,ixcldliq)  ! Grid-ave all cloud liquid
+    allcld_ice(:ncol,top_lev:pver) = state%q(:ncol,top_lev:pver,ixcldice)  !           "        ice
+  end if
+
+  ! ------------------------------------------------------------ !
+  ! Compute in cloud ice and liquid mixing ratios                !
+  ! Note that 'iclwp, iciwp' are used for radiation computation. !
+  ! ------------------------------------------------------------ !
+  do k = 1,pver
+    do i = 1,ncol
+      iciwp(i,k)  = 0._r8
+      iclwp(i,k)  = 0._r8
+      ! icimr(i,k)  = 0._r8
```

Comment:
These should go away if we aren't using them.

Reply:
will do